cudax/include/cuda/experimental/__memory_resource/async_memory_pool.cuh

```diff
@@ -64,6 +64,33 @@ inline void __device_supports_stream_ordered_allocations(const int __device_id)
   }
 }
 
+inline void
+__mempool_switch_peer_access(cudaMemPool_t __pool, const device_ref* __devices, size_t __count, bool __enable)
+{
+  ::std::vector<cudaMemAccessDesc> __descs;
+  __descs.reserve(__count);
+  cudaMemAccessDesc __desc;
+  __desc.flags         = __enable ? cudaMemAccessFlagsProtReadWrite : cudaMemAccessFlagsProtNone;
+  __desc.location.type = cudaMemLocationTypeDevice;
+  for (size_t __i = 0; __i < __count; ++__i)
+  {
+    __desc.location.id = __devices[__i].get();
+    __descs.push_back(__desc);
+  }
+  _CCCL_TRY_CUDA_API(
+    ::cudaMemPoolSetAccess, "Failed to set access of a memory pool", __pool, __descs.data(), __descs.size());
```

Comment:
Could this rather be:
```suggestion
  const cudaMemAccessFlags __flags = __enable ? cudaMemAccessFlagsProtReadWrite : cudaMemAccessFlagsProtNone;
  ::std::vector<cudaMemAccessDesc> __descs(__count, {__flags, {0, cudaMemLocationTypeDevice}});
  ::cuda::std::tranform(__devices.begin(), __devices.end(), __desc.begin(), __set_location_id());
  _CCCL_TRY_CUDA_API(
    ::cudaMemPoolSetAccess, "Failed to set access of a memory pool", __pool, __descs.data(), __descs.size());
```

Reply:
Played with that and it seems worse to do it with algorithms

Reply:
Yeah, I had it implemented with a transform first, but I ended up changing it to a simple loop, seemed a bit better